lib/oj/bag.rb

```diff
@@ -50,13 +49,11 @@ def eql?(other)
       ova = other.instance_variables
       iv = instance_variables
       return false if ova.size != iv.size
-      iv.each do |vid|
-        return false if instance_variable_get(vid) != other.instance_variable_get(vid)
-      end
+      return false if iv.any? { |vid| instance_variable_get(vid) != other.instance_variable_get(vid) }
       true
```

Comment:
I think it's better to use `all?` method then. http://ruby-doc.org/core-1.8.7/Enumerable.html#method-i-all-3F

Agreed?